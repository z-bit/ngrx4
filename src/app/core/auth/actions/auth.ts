import { Action } from '@ngrx/store';
import { User } from '../models/user';
import { Authenticate } from '../models/authenticate';

export const LOGIN = '[Auth] Login';
export const LOGIN_SUCCESS = '[Auth] LOgin Success';
export const LOGIN_FAILURE = '[Auth] Login Failure';
export const LOGIN_REDIRECT = '[Auth] Login Redirect';
export const LOGOUT = '[Auth] Logout';

export class Login implements Action {
  readonly type = LOGIN;
  constructor(
    public payload: Authenticate
  ) {}
}

export class LoginSuccess implements Action {
  readonly type = LOGIN_SUCCESS;
  constructor(
    public payload: { user: User }
  ) {}
}

export class LoginFailure implements Action {
  readonly type = LOGIN_FAILURE;
  constructor(
    public payload: any
  ) {}
}

export class LoginRedirect implements Action {
  readonly type = LOGIN_REDIRECT;
}

export class Logout implements Action {
  readonly type = LOGOUT;
}

export type Actions
  = Login
  | LoginSuccess
  | LoginFailure
  | LoginRedirect
  | Logout
;
