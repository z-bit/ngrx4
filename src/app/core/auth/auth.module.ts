import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { LoginFormComponent } from './components/login-form/login-form.component';


@NgModule({
  imports: [
    CommonModule
  ],
  declarations: [
    LoginFormComponent,
  ]
})
export class AuthModule { }
