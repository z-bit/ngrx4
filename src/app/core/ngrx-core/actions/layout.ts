import { Action } from '@ngrx/store';
export interface layoutAction extends Action {
  type: string;
  payload?: any;
}

export const OPEN_SIDENAV = '[Layout] Open Sidenav';
export const CLOSE_SIDENAV = '[Layout] Close Sidenav';

export class OpenSidenav implements layoutAction {
  readonly type = OPEN_SIDENAV;
}

export class CloseSidenav implements layoutAction {
  readonly type = CLOSE_SIDENAV;
}

export type Actions
  = OpenSidenav
  | CloseSidenav
;
