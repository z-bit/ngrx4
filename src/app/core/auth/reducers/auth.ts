import * as authActions from '../actions/auth';
import { User } from '../models/user';

export interface State {
  loggedIn: boolean;
  user: User | null;
}

export const initialState: State = {
  loggedIn: false,
  user: null,
};

export function reducer(state = initialState, action: authActions.Actions) {
  switch (action.type) {
    case authActions.LOGIN_SUCCESS: {
      return {
        ...state,
        loggedIn: true,
        user: action.payload.user,
      };
    }
    case authActions.LOGOUT: {
      return initialState;
    }
    default: {
      return state;
    }
  }
}

export const getLoggedIn = (state: State) => state.loggedIn;
export const getUser = (state: State) => state.user;
