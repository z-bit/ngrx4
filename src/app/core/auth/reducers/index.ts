import { createSelector, createFeatureSelector } from '@ngrx/store';
import * as rootReducers from '../../reducers';
import * as authReducers from './auth';
import * as loginPageReducers from './login-page';

export interface AuthState {
  status: authReducers.State;
  loginPage: loginPageReducers.State;
}

export interface State extends rootReducers.State {
  auth: AuthState;
}

export const reducers = {
  status: authReducers.reducer,
  loginPage: loginPageReducers.reducer,
};

export const selectAuthState = createFeatureSelector<AuthState>('auth');

export const selectAuthStatusState = createSelector(
  selectAuthState,
  (state: AuthState) => state.status
);
export const getLoggedIn = createSelector(
  selectAuthStatusState,
  authReducers.getLoggedIn
);
export const getUser = createSelector(
  selectAuthStatusState,
  authReducers.getUser
);
export const selectLoginPageState = createSelector(
  selectAuthState,
  (state: AuthState) => state.loginPage
);
export const getLoginPageError = createSelector(
  selectLoginPageState,
  loginPageReducers.getError
);
export const getLoginPagePending = createSelector(
  selectLoginPageState,
  loginPageReducers.getPending
);

