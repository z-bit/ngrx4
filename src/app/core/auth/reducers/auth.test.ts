import { reducer } from './auth';
import * as authReducers from './auth';
import { Login, LoginSuccess, Logout } from '../actions/auth';
import { User } from '../models/user';
import { Authenticate } from '../models/authenticate';

describe('AuthReducer', () => {
  describe('undefined action', () => {
    it('should return state', () => {
      const action = {} as any;
      const result = reducer(undefined, action);
      expect(result).toMatchSnapshot();
    });
  });
});


