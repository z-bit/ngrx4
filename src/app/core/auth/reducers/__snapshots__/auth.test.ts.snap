// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthReducer undefined action should return state 1`] = `
Object {
  "loggedIn": false,
  "user": null,
}
`;
